openapi.yaml

```diff
@@ -171,6 +547,365 @@ components:
+    BookAttributes:
+      allOf:
+        - $ref: '#/components/schemas/BookPostAttributes'
+        - type: object
+          properties:
+            available:
+              $ref: '#/components/schemas/BookAvailable'
+    BookResource:
+      type: object
+      properties:
+        id:
+          $ref: '#/components/schemas/BookId'
+        type:
+          $ref: '#/components/schemas/BookType'
+        links:
+          $ref: '#/components/schemas/SelfLink'
+        attributes:
+          $ref: '#/components/schemas/BookAttributes'
+    BookResult:
+      properties:
+        links:
+            $ref: '#/components/schemas/SelfLink'
+        data:
+            $ref: '#/components/schemas/BookResource'
+    BookSetResult:
+      properties:
+        links:
+            $ref: '#/components/schemas/SelfLink'
+        data:
+          type: array
+          items:
+            $ref: '#/components/schemas/BookResource'
+    BookPatchBody:
+      type: object
+      required:
+        - data
+      properties:
+        data:
+          type: object
+          required:
+            - id
+            - type
+          additionalProperties: false
+          properties:
+            id:
+              $ref: '#/components/schemas/BookId'
+            type:
+              $ref: '#/components/schemas/BookType'
+            attributes:
+              $ref: '#/components/schemas/BookAttributes'
+    MemberPostBody:
+      type: object
+      required:
+        - data
+      additionalProperties: false
+      properties:
+        data:
+          type: object
+          required:
+            - type
+            - attributes
+          additionalProperties: false
+          properties:
+            type:
+              $ref: '#/components/schemas/MemberType'
+            attributes:
+              allOf:
+                - $ref: '#/components/schemas/MemberPostRequired'
+                - $ref: '#/components/schemas/MemberPostAttributes'
+    MemberPostRequired:
+      type: object
+      required:
+        - firstName
+        - lastName
+        - email
+        - address
+        - phoneNumber
+    MemberPostAttributes:
+      type: object
+      properties:
+        firstName:
+          type: string
+          nullable: false
+          description: First name of member
+          example: John
+        lastName:
+          type: string
+          nullable: false
+          description: Last name of member
+          example: Doe
+        email:
+          type: string
+          nullable: false
+          description: Email of member
+          example: john.doe@example.com
+        address:
+          type: string
+          nullable: false
+          description: Address of member
+          example: 123 Main Street, Cityville
+        phoneNumber:
+          type: string
+          nullable: false
+          description: Phone number of member
+          example: 555-123-4567
```

Comment:
You should add `pattern` for this field as well.

Reply:
done